rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JIRI.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+/**
+ * Marker interface for RDF4J implementations of IRI.
+ * 
+ */
+public interface RDF4JIRI 
+	extends RDF4JTerm<org.eclipse.rdf4j.model.IRI>, org.apache.commons.rdf.api.IRI {
+	
+	/**
+	 * Return the corresponding RDF4J {@link org.eclipse.rdf4j.model.IRI}.
+	 * 
+	 * @return The corresponding RDF4J IRI
+	 */
+	public org.eclipse.rdf4j.model.IRI asValue();
```

Comment:
`RDF4JIRI` is not an instance of `org.eclipse.rdf4j.model.IRI` **and** `org.apache.commons.rdf.api.IRI` - however it's a `RDF4JTerm` specialization with the rdf4j `IRI` as its [encapsulated field](https://github.com/apache/incubator-commonsrdf/blob/06711f9c2832609680b9775de39456b5b2b71ccb/rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/impl/AbstractRDFTerm.java).

It's an interesting idea to make `RDF4JIRI` implement both interfaces - but then we would be violating the `false` definition for Commons RDF [IRI.equals()](http://stain.github.io/incubator-commonsrdf/COMMONSRDF-35/org/apache/commons/rdf/api/IRI.html#equals-java.lang.Object-) and [RDF4J IRI.equals()](http://rdf4j.org/javadoc/2.0/org/eclipse/rdf4j/model/IRI.html#equals-java.lang.Object-) -- perhaps both definitions could be loosened to only define the `true` case and be open-ended - but it means you could have:

```java
org.eclipse.rdf4j.model.IRI a; // from regular RDF4j
org.apache.commons.rdf.rdf4j.RDF4JIRI b; // from RDF4JTermFactory
org.apache.commons.rdf.api.IRI c; // from JenaRDFTermFactory
```

and equalities:
```java
assert a.equals(b);
assert b.equals(a);
assert b.equals(c);
assert c.equals(b);
```

but confusingly NOT

```java
org.eclipse.rdf4j.model.IRI a; // from regular RDF4j
org.apache.commons.rdf.rdf4j.RDF4JIRI b; // from RDF4JTermFactory
org.apache.commons.rdf.api.IRI c; // from JenaRDFTermFactory

assert a.equals(c);
assert c.equals(a);
```

which could lead to fun behaviour in say a HashMap..

To support it properly I think we would need to make `org.eclipse.rdf4j.model.IRI` extend `org.apache.commons.rdf.api.IRI` directly.